### PR TITLE
[VAN-753] Change GEP style for array constraints

### PR DESCRIPTION
**Pull request by 0xddom**

### CODE REVIEW

File: code_producers/src/llvm_elements/instructions.rs [hunk 1]
```diff
@@ -860,3 +860,18 @@ pub fn create_switch<'a>(
 ) -> InstructionValue<'a> {
     producer.builder().build_switch(value, else_block, cases)
 }
+
+pub fn get_index_from_gep(gep: PointerValue) -> usize {
+    let op = gep
```
Comment (tim-hoffman) [inline code_producers/src/llvm_elements/instructions.rs:865]: Maybe it should assert that `gep` has the GEP opcode.

Comment (tim-hoffman) [reply code_producers/src/llvm_elements/instructions.rs:865]: Maybe also assert that operand 1 is `0`?

File: code_producers/src/llvm_elements/instructions.rs [hunk 2]
```diff
@@ -860,3 +860,18 @@ pub fn create_switch<'a>(
...
+        .get_operand(2);
+    let idx = op
+        .expect("Second index is missing in GEP that is meant to be a signal")
+        .expect_left("Second index must be a basic value!");
+    let n = match idx {
+        BasicValueEnum::IntValue(v) => v.get_sign_extended_constant(),
+        _ => panic!("Second index must be an integer value!")
+    };
+    n.expect("Could not load the integer value of the IntValue!") as usize
+}
```
Comment (tim-hoffman) [inline code_producers/src/llvm_elements/instructions.rs:877]: nit: missing newline at end of file

**Review by tim-hoffman — COMMENTED** [review]
I left a few suggestions.

**Comment by 0xddom** [comment]
Looking at the failing test case I think this may not work with subcomponent signals but Im not sure yet

**Comment by 0xddom** [comment]
Yeah, I'm creating GEPs from %0 when one of them should be from %6 (in this case)

File: code_producers/src/llvm_elements/instructions.rs (hunk 1 above)
Comment (tim-hoffman) [reply code_producers/src/llvm_elements/instructions.rs]: That's exactly why I would add the assert. If some weird case ever appeared where it wasn't 0 for some reason, we would want to know.

**Review by tim-hoffman — APPROVED** [review]
LGTM

**Pull request merged**